test/service/ConfigLoader.test.ts

```diff
@@ -43,32 +62,153 @@ describe('ConfigLoader', () => {
+        expect(upgradedPresetData.customPresetCache).deep.eq({
+            maxUnlockedAccounts: 30,
+            nodes: [
+                {
+                    host: 'my-host.io',
+                    mainPrivateKey: 'CA82E7ADAF7AB729A5462A1BD5AA78632390634904A64EB1BB22295E2E1A1BDD',
+                    remotePrivateKey: 'EFE3F0EF0AB368B8D7AC194D52A8CCFA2D5050B80B9C76E4D2F4D4BF2CD461C1',
+                    transportPrivateKey: '6154154096354BC3DB522174ACD8BFE553893A0991BD5D105599846F17A3383B',
+                    voting: true,
+                    vrfPrivateKey: 'F3C24C153783B683E40FB2671493B54480370BF4E3AB8027D4BF1293E14EB9B8',
+                },
+            ],
+            privateKeySecurityMode: 'PROMPT_MAIN',
+        });
+
+        const upgradedPresetResetToDefaults = await configLoader.createPresetData({
+            oldPresetData: originalPresetData,
+            preset: Preset.testnet,
+            customPresetObject: {
+                maxUnlockedAccounts: 15,
+            },
+            assembly: Assembly.dual,
+            password: 'abcd',
+        });
+        expect(upgradedPresetResetToDefaults).to.not.be.undefined;
+        expect(upgradedPresetResetToDefaults.nodes![0].maxUnlockedAccounts).eq(undefined);
+        expect(upgradedPresetResetToDefaults.maxUnlockedAccounts).eq(15);
+        expect(upgradedPresetResetToDefaults.customPresetCache).deep.eq({
+            maxUnlockedAccounts: 15,
+        });
+    });
+
+    it('mergePreset', async () => {
+        const configLoader = new ConfigLoaderMocked(logger);
+        expect(
+            configLoader.mergePresets(
+                { maxUnlockedAccounts: 1, inflation: { a: 1, c: 1, d: 1 } },
+                { maxUnlockedAccounts: 2 },
+                { maxUnlockedAccounts: 3, inflation: { c: 2, d: 2, e: 2 } },
+                { maxUnlockedAccounts: 4 },
+            ),
+        ).deep.eq({ maxUnlockedAccounts: 4, inflation: { c: 2, d: 2, e: 2 } });
+    });
+
+    it('mergePreset with node', async () => {
+        const configLoader = new ConfigLoaderMocked(logger);
+        const merged = configLoader.mergePresets(
+            {
+                maxUnlockedAccounts: 1,
+                inflation: { a: 1, c: 1, d: 1 },
+            },
+            { nodes: [{ maxUnlockedAccounts: 5, name: 'name1' }], knownRestGateways: ['r1', 'r2'] },
+            { maxUnlockedAccounts: 3, inflation: { c: 2, d: 2, e: 2 }, knownRestGateways: ['r2', 'r3'] },
+            {
+                maxUnlockedAccounts: 4,
+                nodes: [{ maxUnlockedAccounts: 3 }, { maxUnlockedAccounts: 4, name: 'nameB' }],
+            },
+        );
+        expect(merged).deep.eq({
+            maxUnlockedAccounts: 4,
+            inflation: { c: 2, d: 2, e: 2 },
+            nodes: [
+                { maxUnlockedAccounts: 3, name: 'name1' },
+                { maxUnlockedAccounts: 4, name: 'nameB' },
+            ],
+            knownRestGateways: ['r2', 'r3'],
+        });
```

Comment:
while beyond the scope of this change, longer term, it would probably be better to refactor the configuration merging into a self contained file/class instead of having it buried in the ConfigLoader  (the fact that to test you need to use a ConfigLoaderMocked is a smell). this would be more self contained / single purpose and be easier to test and maintain.

Reply:
Yes, ConfigLoader, BootstrapUtils, and maybe a few others didn't grow as nicely as I would want.  I need to split, re-architecture, and refactor.